### ISSUE TITLE
[Feature Request]: Waybar Select Menu similar to Rofi Style Select Menu

### ISSUE THREAD

**Issue by saulback** [prasanthrangan/hyprdots#1300]
### Description

MOD + Shift + B (or other key combination) will open a waybar select menu where you can see previews of all the possible waybar options, similar to how wallpapers, themes, rofi all have a select menu. User can than select the desired waybar.

### Use Case

Currently to select the desired waybar layout, you have to MOD + ALT + down/up until you find the one you are looking for. Each time selecting the new waybar takes some time. Looking at the default config.ctl, there are 24 different options that you would need to go through. So, especially for a beginner, they would have to navigate through all 24 to identify the one they liked most, then keep navigating through to finally land on the one they wanted.

A menu would make this process drastically easier. A user would just open the menu and be able to see all the possible options and directly select the one they want without having to cycle through every one.

### Proposed Solution

First, it seems to me that there probably aren't 24 distinct layouts. There are probably around 6 different layouts that go either top, bottom, right, left. So I think we could have 2 sections of the menu, 1 where you select where you want the waybar - top, left, right or bottom. Then another section where you select from the 6-ish layouts.

In the config.ctl, we could specify the preview image to use for each line. The files would live in a preview folder in the same directory. 

### Alternatives

I haven't considered alternatives.

### Additional Information

[Add any other information about the feature request here. For example, you might include links to similar features in other projects, or screenshots or diagrams to help explain your idea.]
For examples, see the Wallpaper/Rofi/Theme selector menus.

**Comment by kRHYME7** [comment]
Should be good, but we should take note that these layouts can be changed/added/deleted by the user having a preferred layout.

**Comment by dieBakterie** [comment]
Yeah that's an amazing Idea.

**Comment by dieBakterie** [comment]
We can Go them through the config.ctl and doing Like an diaoshow?

**Comment by prasanthrangan** [comment]
Hi, this is possible in theory...

> First, it seems to me that there probably aren't 24 distinct layouts. There are probably around 6 different layouts that go either top, bottom, right, left. So I think we could have 2 sections of the menu, 1 where you select where you want the waybar - top, left, right or bottom. Then another section where you select from the 6-ish layouts.

the layouts in `config.ctl` was just to show the possibilities, and its not predefined... users can add as many layouts as they can think of, with different modules/order/position/size/grouping as they like.

> In the config.ctl, we could specify the preview image to use for each line. The files would live in a preview folder in the same directory.

this would be a nightmare to create an image for each line, so the only way is to make it dynamic.
but to make the previews dynamic and modular, each module should have its own `svg`/`png` (the hardest part).
then it will be easy to group these images based on the layouts/lines in `config.ctl` to show a preview for each line.

**Comment by dieBakterie** [comment]
Yeah sorry haven't thought how complex and hard to implement this would be.

**Comment by prasanthrangan** [comment]
> Yeah sorry haven't thought how complex and hard to implement this would be.

No worries, I think it can be done...
But first, need to have `svg` images for each available (25) modules, to generate previews each layout.

```
backlight
battery
bluetooth
cliphist
clock
cpu
cpuinfo
gpuinfo
idle_inhibitor
keybindhint
language
memory
mpris
network
power
pulseaudio
spotify
taskbar
theme
tray
updates
wallchange
wbar
window
workspaces
```

**Comment by rubiin** [comment]
we can also include the default svgs for waybar on the readme too

**Comment by its-19818942118** [comment]
I think adding this for lock screen or sddm would be great. But it will be too much work though

**Comment by Sitolam** [comment]
I would like to see a dynamic waybar maker 😂
